etw/rust/tracelogging_macros/src/provider_generator.rs

```diff
@@ -50,6 +50,7 @@ impl ProviderGenerator {
         let prov_tokens = self
             .prov_tree
             // static PROVIDER: ::tracelogging::Provider = unsafe { ... };
+            .add_ident("pub")
```

Comment:
This probably should be in a different PR.

Reply:
Making this pub is a bit problematic since the provider must be strictly scoped to the crate and must not be exported from the crate. My idea is that you would declare the provider in the root module so that child modules can use it.

Reply:
The purpose of making this `pub` was that it was not accessible for different files inside the same crate. 

I was unable to access this static in my `device.rs` after I declared it in my `driver.rs`. The error I received was as follows:

```
device.rs 1 of 1 problem
static `MY_TRACELOGGING_PROVIDER` is private
private static rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics)
driver.rs(9, 1): the static `MY_TRACELOGGING_PROVIDER` is defined here
```

in the code `use driver::MY_TRACELOGGING_PROVIDER;`

Reply:
Making it pub causes problems. Can you instead put your provider in lib.rs or in some other parent mod so that it doesn't need to be public?

Reply:
Removed pub ident